### PR TITLE
Create files for main gulp tasks

### PR DESCRIPTION
**Pull request by felixzapata**
Use the tasks folder to create files for the main tasks of the gulpfile. 

This way I think it is more clear.

### CODE REVIEW

File: tasks/cache-config.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      var md5 = crypto.createHash('md5');
+      md5.update(JSON.stringify(config.precache));
+      config.precacheFingerprint = md5.digest('hex');
+
+      var configPath = path.join(dir, 'cache-config.json');
+      fs.writeFile(configPath, JSON.stringify(config), callback);
+    }
+  });
+});
+
```
Comment (abdonrd) [inline tasks/cache-config.js]: I think that we can remove this line. :)

File: tasks/clean.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+var path = require('path');
+var dist = function(subpath) {
+  return !subpath ? DIST : path.join(DIST, subpath);
+};
+
+// Clean output directory
+gulp.task('clean', function() {
+  return del(['.tmp', dist()]);
+});
+
```
Comment (abdonrd) [inline tasks/clean.js]: I think that we can remove these lines. :)

File: tasks/copy.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  // These are things which cannot be vulcanized
+  var bower = gulp.src([
+    'app/bower_components/{webcomponentsjs,platinum-sw,sw-toolbox,promise-polyfill}/**/*'
+  ]).pipe(gulp.dest(dist('bower_components')));
+
+  return merge(app, bower)
+    .pipe($.size({
+      title: 'copy'
+    }));
+});
```
Comment (abdonrd) [inline tasks/copy.js]: Don't remove the last line, please. :)

File: tasks/deploy.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+gulp.task('deploy-gh-pages', function() {
+  return gulp.src(dist('**/*'))
+    // Check if running task from Travis CI, if so run using GH_TOKEN
+    // otherwise run using ghPages defaults.
+    .pipe($.if(process.env.TRAVIS === 'true', $.ghPages({
+      remoteUrl: 'https://$GH_TOKEN@github.com/polymerelements/polymer-starter-kit.git',
+      silent: true,
+      branch: 'gh-pages'
+    }), $.ghPages()));
+});
```
Comment (abdonrd) [inline tasks/deploy.js]: Don't remove the last line, please. :)

File: tasks/fonts.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+var gulp = require('gulp');
+var $ = require('gulp-load-plugins')();
+var DIST = 'dist';
+var path = require('path');
+var dist = function(subpath) {
+  return !subpath ? DIST : path.join(DIST, subpath);
+};
+
+
```
Comment (abdonrd) [inline tasks/fonts.js]: I think that we can remove this line. :)

File: tasks/html.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+      spare: true
+    })))
+    // Output files
+    .pipe(gulp.dest(dest))
+    .pipe($.size({
+      title: 'html'
+    }));
+};
+
+
```
Comment (abdonrd) [inline tasks/html.js]: I think that we can remove this line. :)

File: tasks/images.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      interlaced: true
+    }))
+    .pipe(gulp.dest(dest))
+    .pipe($.size({title: 'images'}));
+};
+
+// Optimize Images
+gulp.task('images', function() {
+  return imageOptimizeTask('app/images/**/*', dist('images'));
+});
```
Comment (abdonrd) [inline tasks/images.js]: Don't remove the last line, please. :)

File: tasks/styles.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+var $ = require('gulp-load-plugins')();
+var path = require('path');
+var browserSync = require('browser-sync');
+var reload = browserSync.reload;
+var DIST = 'dist';
+var dist = function(subpath) {
+  return !subpath ? DIST : path.join(DIST, subpath);
+};
+
+
```
Comment (abdonrd) [inline tasks/styles.js]: I think that we can remove this line. :)

File: tasks/styles.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    }))
+    .pipe($.changed(stylesPath, {extension: '.css'}))
+    .pipe($.autoprefixer(AUTOPREFIXER_BROWSERS))
+    .pipe(gulp.dest('.tmp/' + stylesPath))
+    .pipe($.minifyCss())
+    .pipe(gulp.dest(dist(stylesPath)))
+    .pipe($.size({title: stylesPath}));
+};
+
+
```
Comment (abdonrd) [inline tasks/styles.js]: I think that we can remove this line. :)

File: tasks/styles.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+// Compile and automatically prefix stylesheets
+gulp.task('styles', function() {
+  return styleTask('styles', ['**/*.css']);
+});
+
+gulp.task('elements', function() {
+  return styleTask('elements', ['**/*.css']);
+});
+
```
Comment (abdonrd) [inline tasks/styles.js]: I think that we can remove this line. :)

File: tasks/watch.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
+'use strict';
```
Comment (abdonrd) [inline tasks/watch.js]: I think that we can add one line here. :)

File: tasks/watch.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+
+// Watch Files For Changes
+gulp.task('watch', function () {
+  gulp.watch(['app/**/*.html', '!app/bower_components/**/*.html'], reload);
+  gulp.watch(['app/styles/**/*.css'], ['styles', reload]);
+  gulp.watch(['app/elements/**/*.css'], ['elements', reload]);
+  gulp.watch(['app/scripts/**/*.js'], reload);
+  gulp.watch(['app/images/**/*'], reload);
+});
+
```
Comment (abdonrd) [inline tasks/watch.js]: I think that we can remove this line. :)

**Comment by abdonrd** [comment]
Comments with minor changes, but LGTM! Great work, @felixzapata! :tada:

**Comment by robdodson** [comment]
Hey @felixzapata thanks for the work on this. We're currently planning a large refactor of PSK that will include breaking up the gulpfile. We're currently at a team offsite discussing what this refactor will look like. We plan to put out a blog post shortly to lay out the plan.

In the meantime I don't think we will be able to merge this because it will be a major version change and will also require updating all of the recipes in the `docs/` directory. We'll be sure to keep you looped in because we'd love your help in working on this.

**Comment by felixzapata** [comment]
ook

**Comment by samccone** [comment]
:+1: ditto to what rob said.

@felixzapata really really really appreciate this exploration and PR, it totally has helped us to shape the direction for the new tools that are being worked on.

I will ping you on the repo as soon as it is on github. I would :heart: your input.

Thanks!!

**Pull request closed without merging**